app/pages/project/index.cjsx

```diff
@@ -27,164 +35,222 @@ SOCIAL_ICONS =
+              <img src={@state.avatar.src} className="avatar" />}
+            Visit {@props.project.display_name}
+          </a>
+        else
+          <IndexLink to="#{projectPath}" activeClassName="active" className="tabbed-content-tab">
+            {if @state.avatar?
+              <img src={@state.avatar.src} className="avatar" />}
+            {@props.project.display_name}
+          </IndexLink>}
+
+        {unless @props.project.redirect
+          <Link to="#{projectPath}/research" activeClassName="active" className="tabbed-content-tab">
+            <Translate content="project.nav.research" />
+          </Link>}
+
+        {if @props.project.redirect
+          <a href={@redirectClassifyLink(@props.project.redirect)} className="tabbed-content-tab" target="_blank">
+            <Translate content="project.nav.classify" />
+          </a>
+        else
+          <Link to="#{projectPath}/classify" activeClassName="active" className="classify tabbed-content-tab">
+            <Translate content="project.nav.classify" />
+          </Link>}
+
+        {if !!pages.results?.content
+          <Link to="#{projectPath}/results" activeClassName="active"className="tabbed-content-tab">
+            {pages.results.title}
+          </Link>}
+
+        {if !!pages.faq?.content
+          <Link to="#{projectPath}/faq" activeClassName="active" className="tabbed-content-tab">
+            {pages.faq.title}
+          </Link>}
+
+        {if !!pages.education?.content
+          <Link to="#{projectPath}/education" activeClassName="active" className="tabbed-content-tab">
+            {pages.education.title}
+          </Link>}
+
+        <Link to="#{projectPath}/talk" activeClassName="active" className="tabbed-content-tab">
+          <Translate content="project.nav.talk" />
+        </Link>
+
+        {@props.project.urls.map ({label, url}, i) =>
+          unless !!label
+            for pattern, icon of SOCIAL_ICONS
+              if url.indexOf(pattern) isnt -1
+                iconForLabel = icon
+            iconForLabel ?= 'globe'
+            label = <i className="fa fa-#{iconForLabel} fa-fw fa-2x"></i>
+          <a key={i} href={url} className="tabbed-content-tab" target="#{@props.project.id}#{url}">{label}</a>}
+      </nav>
+
+      {if !!@props.project.configuration?.announcement
+        <div className="informational project-announcement-banner">
+          <Markdown>{@props.project.configuration.announcement}</Markdown>
+        </div>}
+
+      {React.cloneElement @props.children, user: @props.user, project: @props.project, owner: @props.owner}
+
+      {unless @props.project.launch_approved or @props.project.beta_approved
+        <Translate component="p" className="project-disclaimer" content="project.disclaimer" />}
+
+      <PotentialFieldGuide project={@props.project} />
+    </div>
+
+
+ProjectPageController = React.createClass
+  displayName: 'ProjectPageController'
+
+  mixins: [TitleMixin]
 
   title: ->
```

Comment:
`TitleMixin` calls this to update the document title. (`TitleMixin` will eventually be replaced with a wrapper component.)